src/inputs/pcap/PcapInputStream.cpp

```diff
@@ -319,6 +319,10 @@ void PcapInputStream::_generate_mock_traffic()
 
 void PcapInputStream::process_raw_packet(pcpp::RawPacket *rawPacket)
 {
+    if (processing_packet) {
```

Comment:
used a atomic bool in this case because it will have less overhead then a mutex lock and try lock